api/auth.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+package envoy.api.v2;
+
+message AuthAction {
+
+  enum VerifyType {
+    option allow_alias = true;
+
+    DEFAULT = 0;
+    // Request matches all rules are allowed, otherwise denied.
+    ALLOW = 0;
+    // Request matches all rules or missing required auth fields are denied,
+    // otherwise allowed.
+    DENY = 1;
+  }
+
+  VerifyType verify_type = 1;
+
+  // Check peer identity using X.509 certificate.
+  message X509 {
```

Comment:
+1 on some discussion of how much of `CertificateValidationContext` can be reused as @PiotrSikora suggested. I agree that not everything will be useful in a route action, and we're somewhat bound by not making breaking changes at this point due to the API being frozen for LDS/CDS. It might be fine as is, but I'd like to make sure there isn't anything being missed. 

As a contrived example, in the same way in which I would want to verify a particular route as having a particular SAN, might I not always want to ensure I reached the route via a listener that performed validation via a specific CA? It sounds contrived to me and not very realistic, but other folks probably have opinions.